### PR TITLE
cli: use FlagInfo in CCL code.

### PR DESCRIPTION
**Pull request by mberhault**
* move some of the FlagInfo helpers to `cliflags`
* use a similar flag definition layout in cliccl
* add `StringSliceFlag`
* convert all ccl flags to `FlagInfo`
* use the `FlagInfo` name for the encryption flag

I'm leaving the `cli.<type>Flag` functions in `cli` to avoid having to
prepend `cliflags.` to all of them.

### CODE REVIEW

File: pkg/ccl/baseccl/encryption_spec.go [hunk 1]
```diff
@@ -127,8 +128,7 @@ var _ pflag.Value = &StoreEncryptionSpecList{}
 func (encl StoreEncryptionSpecList) String() string {
 	var buffer bytes.Buffer
 	for _, ss := range encl.Specs {
-		// TODO(mberhault): use the FlagInfo once refactored.
-		fmt.Fprintf(&buffer, "--enterprise-encryption=%s ", ss)
+		fmt.Fprintf(&buffer, "--%s=%s ", cliflagsccl.EnterpriseEncryption.Name, ss)
```
Comment (mberhault) [inline pkg/ccl/baseccl/encryption_spec.go]: @tschottdorf: I think that's the line you wanted. So much work extra stuff for that one line.

**Comment by mberhault** [comment]
+ @mjibson for load csv

**Comment by mjibson** [comment]
CSV stuff LGTM.

**Comment by tschottdorf** [comment]
:lgtm:, thanks!

---

Reviewed 6 of 7 files at r1, 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion, all commit checks successful.

---

*[pkg/ccl/baseccl/encryption_spec.go, line 131 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/20358#-L-DPS4rOvc5AzSyFNPN-r1-131:-L-DkPRcjd7yl4nDI3zL:b-rsdbzm) ([raw file](https://github.com/cockroachdb/cockroach/blob/6d809e4941eb17492d2c180a95a2bf4c4b73313b/pkg/ccl/baseccl/encryption_spec.go#L131)):*
<details><summary><i>Previously, mberhault (marc) wrote…</i></summary><blockquote>

@tschottdorf: I think that's the line you wanted. So much work extra stuff for that one line. 
</blockquote></details>

Welp, that is a lot of work for one line. But what a beautiful line it is :crying_cat_face: 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/20358#-:-L-Dkj0erKo53Y1NNMdu:bpqxd5f)*

**Pull request merged**